### ISSUE TITLE
reduce_euclidean_norm false results

### ISSUE THREAD

**Issue by martinmamql** [tensorflow/tensorflow#36554]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Google colab
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): Google colab
- TensorFlow version (use command below): 1.15.0
- Python version: 3.6.9
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: 10.1.243
- GPU model and memory: N/A

**Describe the current behavior**
reduce_euclidean_norm might return false results?
For example, the following three snippets are all using reduce_euclidean_norm, but all return wrong results (return 4; should return sqrt(17)).

x = tf.constant([[1, 2, 3], [1, 1, 1]])
y = tf.math.reduce_euclidean_norm(x)
with tf.Session() as sess:  print(y.eval()) 
returns 4, should return numerical value of sqrt(17)

x = tf.constant([[1, 2, 3], [1, 1, 1]])
y = tf.compat.v1.math.reduce_euclidean_norm(x)
with tf.Session() as sess:  print(y.eval()) 
returns 4, should return numerical value of sqrt(17)

x = tf.constant([[1, 2, 3], [1, 1, 1]])
y = tf.compat.v2.math.reduce_euclidean_norm(x)
with tf.Session() as sess:  print(y.eval()) 
returns 4, should return numerical value of sqrt(17)

**Describe the expected behavior**

from tensorflow docs
https://www.tensorflow.org/api_docs/python/tf/math/reduce_euclidean_norm
x = tf.constant([[1, 2, 3], [1, 1, 1]])
tf.reduce_euclidean_norm(x)  # sqrt(17)

**Code to reproduce the issue**
Attached above. 

**Other info / logs**
N/A

Thank you so much!

**Comment by martinmamql** [comment]
Update: When the input is in the form
`x = tf.constant([[1.0, 2.0, 3.0], [1.0, 1.0, 1.0]])`
the reduce_euclidean_norm returns the correct result.

**Comment by martinmamql** [comment]
Credit: @merterm

**Comment by ravikyram** [comment]
I have tried on colab with TF version 1.15, Nightly versions and was able to reproduce the issue.Please, find the gist [here.](https://colab.research.google.com/gist/ravikyram/4dc242a7a9391723df71d17e0bf68aa0/untitled630.ipynb) Thanks!

**Comment by martinmamql** [comment]
Thank you so much!

**Comment by jvishnuvardhan** [comment]
@martinmamql All those examples are correctly returning 4 instead of sqrt(17). The reason is `x.dtype is tf.int32` and the doc clearly says `Returns : The reduced tensor, of the same dtype as the input_tensor.`. However, the document has typo (sqrt(17)) which needs to be corrected. I raised a PR to correct it in `master` and will be updated in the docs after merging the PR. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?
